### PR TITLE
modernize `Alignment/TrackerAlignment`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Modernize `Alignment/TrackerAlignment`:
   * merge headers with plugin implementation
   * transformed legacy `ED`Modules into thread-safe versions
   * move `cout` to thread-safe `MessageLogger` instructions
   * remove commented out code
   * add `fillDescriptions` methods  

#### PR validation:

It compiles, also tested with unit tests.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 1]
```diff
@@ -1,66 +1,104 @@
...
 
-AlignmentPrescaler::~AlignmentPrescaler() {
-  //
-}
+AlignmentPrescaler::~AlignmentPrescaler() = default;
 
 void AlignmentPrescaler::beginJob() {
   //
-  std::cout << "in AlignmentPrescaler::beginJob" << std::flush;
+  edm::LogPrint("AlignmentPrescaler") << "in AlignmentPrescaler::beginJob" << std::flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:82]: I think the `std::flush` is not needed, the `edm::LogPrint` will take care of it, no?

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 2]
```diff
@@ -1,66 +1,104 @@
...
   fpresc_ = new TFile(prescfilename_.c_str(), "READ");
   tpresc_ = (TTree*)fpresc_->Get(presctreename_.c_str());
   tpresc_->BuildIndex("DetId");
   tpresc_->SetBranchStatus("*", false);
   tpresc_->SetBranchStatus("DetId", true);
   tpresc_->SetBranchStatus("PrescaleFactor", true);
   tpresc_->SetBranchStatus("PrescaleFactorOverlap", true);
-  cout << " Branches activated " << std::flush;
+  edm::LogPrint("AlignmentPrescaler") << " Branches activated " << std::flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:90]: I think the `std::flush` is not needed, the `edm::LogPrint` will take care of it, no?

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 3]
```diff
@@ -1,66 +1,104 @@
...
   detid_ = 0;
   hitPrescFactor_ = 99.0;
   overlapPrescFactor_ = 88.0;
 
   tpresc_->SetBranchAddress("DetId", &detid_);
   tpresc_->SetBranchAddress("PrescaleFactor", &hitPrescFactor_);
   tpresc_->SetBranchAddress("PrescaleFactorOverlap", &overlapPrescFactor_);
-  cout << " addressed " << std::flush;
+  edm::LogPrint("AlignmentPrescaler") << " addressed " << std::flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:98]: I think the `std::flush` is not needed, the `edm::LogPrint` will take care of it, no?

**Comment by mmusich** [comment]
please test

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 4]
```diff
@@ -71,25 +109,25 @@ void AlignmentPrescaler::endJob() {
 }
 
 void AlignmentPrescaler::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
-  // std::cout<<"\n\n#################\n### Starting the AlignmentPrescaler::produce ; Event: "<<iEvent.id().run() <<", "<<iEvent.id().event()<<std::endl;
+  LogDebug("AlignmentPrescaler") << "\n\n#################\n### Starting the AlignmentPrescaler::produce ; Event: "
+                                 << iEvent.id().run() << ", " << iEvent.id().event() << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:113]: Please remove `std::endl`

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 5]
```diff
@@ -71,25 +109,25 @@ void AlignmentPrescaler::endJob() {
...
 
-  //int npxlhits=0;
+  int npxlhits = 0;
 
   //loop on tracks
   for (std::vector<reco::Track>::const_iterator ittrk = Tracks->begin(), edtrk = Tracks->end(); ittrk != edtrk;
        ++ittrk) {
     //loop on tracking rechits
-    // std::cout << "Loop on hits of track #" << (ittrk - Tracks->begin()) << std::endl;
+    LogDebug("AlignmentPrescaler") << "Loop on hits of track #" << (ittrk - Tracks->begin()) << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:130]: Please remove `std::endl`

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 6]
```diff
@@ -135,10 +173,11 @@ void AlignmentPrescaler::produce(edm::Event& iEvent, const edm::EventSetup& iSet
             tmpflag.SetDetId(hit->geographicalId());
             if (tmpflag.isOverlap())
               isOverlapHit = true;
-            // std::cout<<"~*~*~* Prescale (1D) for module "<<tmpflag.detId().rawId()<<"("<<InValMap[stripclust].detId().rawId() <<") is "<<hitPrescFactor_<<std::flush;
-            //  if(tmpflag.isOverlap())cout<<" (it is Overlap)"<<endl;
-            // else cout<<endl;
-
+            LogDebug("AlignmentPrescaler")
+                << "~*~*~* Prescale (1D) for module " << tmpflag.detId().rawId() << "("
+                << InValMap[stripclust].detId().rawId() << ") is " << hitPrescFactor_ << std::flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:178]: I think the std::flush is not needed, the edm::LogPrint will take care of it, no?

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 7]
```diff
@@ -148,17 +187,18 @@ void AlignmentPrescaler::produce(edm::Event& iEvent, const edm::EventSetup& iSet
             tmpflag.SetDetId(hit->geographicalId());
             if (tmpflag.isOverlap())
               isOverlapHit = true;
-            // std::cout<<"~*~*~* Prescale (2D) for module "<<tmpflag.detId().rawId()<<"("<<InValMap[stripclust].detId().rawId() <<") is "<<hitPrescFactor_<<std::flush;
-            //  if(tmpflag.isOverlap())cout<<" (it is Overlap)"<<endl;
-            // else cout<<endl;
-
+            LogDebug("AlignmentPrescaler")
+                << "~*~*~* Prescale (2D) for module " << tmpflag.detId().rawId() << "("
+                << InValMap[stripclust].detId().rawId() << ") is " << hitPrescFactor_ << std::flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:192]: I think the std::flush is not needed, the edm::LogPrint will take care of it, no?

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 8]
```diff
@@ -148,17 +187,18 @@ void AlignmentPrescaler::produce(edm::Event& iEvent, const edm::EventSetup& iSet
...
+            if (tmpflag.isOverlap())
+              LogDebug("AlignmentPrescaler") << " (it is Overlap)" << endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:194]: Please remove `endl`

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 9]
```diff
@@ -169,15 +209,15 @@ void AlignmentPrescaler::produce(edm::Event& iEvent, const edm::EventSetup& iSet
       //      tmpflag.SetDetId(hit->geographicalId());
 
       if (isOverlapHit) {
-        //cout<<"  DetId="<<tmpdetid<<" is Overlap! "<<flush;
+        LogDebug("AlignmentPrescaler") << "  DetId=" << tmpdetid << " is Overlap! " << flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:212]: I think the `flush` is not needed, the edm::LogPrint will take care of it, no?

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc [hunk 10]
```diff
@@ -169,15 +209,15 @@ void AlignmentPrescaler::produce(edm::Event& iEvent, const edm::EventSetup& iSet
...
         takeit = (float(myrand_->Rndm()) <= overlapPrescFactor_);
       }
       if (!takeit) {
         float rr = float(myrand_->Rndm());
         takeit = (rr <= hitPrescFactor_);
       }
       if (takeit) {  //HIT TAKEN !
-        //cout<<"  DetId="<<tmpdetid<<" taken!"<<flush;
+        LogDebug("AlignmentPrescaler") << "  DetId=" << tmpdetid << " taken!" << flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:220]: I think the `flush` is not needed, the edm::LogPrint will take care of it, no?

File: Alignment/TrackerAlignment/plugins/TkAlCaOverlapTagger.cc [hunk 11]
```diff
@@ -133,12 +142,12 @@ void TkAlCaOverlapTagger::produce(edm::Event& iEvent, const edm::EventSetup& iSe
             //
             AlignmentClusterFlag hitflag(hit->geographicalId());
             hitflag.SetOverlapFlag();
-            // cout<<"Overlap found in SubDet "<<subDet<<"!!!"<<flush;
+            LogDebug("TkAlCaOverlapTagger") << "Overlap found in SubDet " << subDet << "!!!" << std::flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaOverlapTagger.cc:145]: I think the `std::flush` is not needed, the edm::LogPrint will take care of it, no?

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 12]
```diff
@@ -65,19 +61,15 @@ TkAlCaSkimTreeMerger::TkAlCaSkimTreeMerger(const edm::ParameterSet &iConfig)
   maxTOBhits_ = maxhitsSet_.getParameter<int32_t>("TOBmaxhits");
   maxTEChits_ = maxhitsSet_.getParameter<int32_t>("TECmaxhits");
   //anything you want to do for initializing
-  std::cout << "\n\n*** MAX N HITS = " << maxhits_ << std::endl << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "\n\n*** MAX N HITS = " << maxhits_ << std::endl << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:64]: Please remove `std::endl`

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 13]
```diff
@@ -65,19 +61,15 @@ TkAlCaSkimTreeMerger::TkAlCaSkimTreeMerger(const edm::ParameterSet &iConfig)
...
 }
 
 TkAlCaSkimTreeMerger::~TkAlCaSkimTreeMerger() {
-  //default destructor
-  // delete out_;
-  // delete firsttree_;
-
   delete ch_;
-  std::cout << "finished." << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "finished." << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:72]: Please remove std::endl

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 14]
```diff
@@ -86,7 +78,7 @@ void TkAlCaSkimTreeMerger::beginJob() {
 
   //prepare the chain
   ch_ = new TChain(treename_.c_str());
-  std::cout << "The chain contains " << ch_->GetNtrees() << " trees" << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "The chain contains " << ch_->GetNtrees() << " trees" << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:81]: Please remove std::endl

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 15]
```diff
@@ -98,15 +90,15 @@ void TkAlCaSkimTreeMerger::beginJob() {
...
+    LogDebug("TkAlCaSkimTreeMerger") << "Adding " << filename << std::endl;
     ch_->Add(filename.c_str());
     if (first) {
       firstfilename_ = filename;
       first = false;
     }
   }
-  std::cout << "Now the chain contains " << ch_->GetNtrees() << " trees (" << ch_->GetEntries() << " entries)"
-            << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "Now the chain contains " << ch_->GetNtrees() << " trees (" << ch_->GetEntries()
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:100]: Please remove std::endl

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 16]
```diff
@@ -98,15 +90,15 @@ void TkAlCaSkimTreeMerger::beginJob() {
     flist >> filename;
     if (filename.empty())
       continue;
-    //std::cout<<"Adding "<<filename<<std::endl;
+    LogDebug("TkAlCaSkimTreeMerger") << "Adding " << filename << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:93]: Please remove std::endl

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 17]
```diff
@@ -153,26 +145,26 @@ void TkAlCaSkimTreeMerger::beginJob() {
...
-  std::cout << "Nhits in the chain: " << totnhits << std::endl;
-  std::cout << "NOverlaps in the chain: " << totnoverlaps << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "Nhits in the chain: " << totnhits << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "NOverlaps in the chain: " << totnoverlaps << std::endl;
 
   myclock.Stop();
-  std::cout << "Finished beginJob after " << myclock.RealTime() << " s (real time) / " << myclock.CpuTime()
-            << " s (cpu time)" << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "Finished beginJob after " << myclock.RealTime() << " s (real time) / "
+                                   << myclock.CpuTime() << " s (cpu time)" << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:153]: Please remove std::endl

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 18]
```diff
@@ -153,26 +145,26 @@ void TkAlCaSkimTreeMerger::beginJob() {
...
   myclock.Continue();
 }  //end beginJob
 
 // ------------ method called to for each event  ------------
 void TkAlCaSkimTreeMerger::analyze(const edm::Event &, const edm::EventSetup &) {
-  // std::cout<<firsttree_->GetEntries()<<std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << firsttree_->GetEntries() << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:159]: Please remove std::endl

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 19]
```diff
@@ -153,26 +145,26 @@ void TkAlCaSkimTreeMerger::beginJob() {
...
 }  //end analyze
 
 // ------------ method called after having analyzed all the events  ------------
 void TkAlCaSkimTreeMerger::endJob() {
   //address variables in the first tree and in the chain
   TFile *firstfile = new TFile(firstfilename_.c_str(), "READ");
   firsttree_ = (TTree *)firstfile->Get(treename_.c_str());
-  std::cout << "the first tree has " << firsttree_->GetEntries() << " entries" << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "the first tree has " << firsttree_->GetEntries() << " entries" << std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:167]: Please remove std::endl

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 20]
```diff
@@ -235,7 +227,7 @@ void TkAlCaSkimTreeMerger::endJob() {
     firsttree_->GetEntry(mod);
     nhits_out = hitmap_[id];
     noverlaps_out = overlapmap_[id];
-    // if(mod<25)std::cout<<"Nhits 1st tree: "<<nhits<<"\tTotal nhits chain: "<<nhits_out<<std::endl;
+    // if(mod<25)LogDebug("TkAlCaSkimTreeMerger")<<"Nhits 1st tree: "<<nhits<<"\tTotal nhits chain: "<<nhits_out<<std::endl;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:230]: Please remove this commented out line if it is not needed

File: Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc [hunk 21]
```diff
@@ -286,16 +278,35 @@ void TkAlCaSkimTreeMerger::endJob() {
...
   myclock.Stop();
-  std::cout << "Finished endJob after " << myclock.RealTime() << " s (real time) / " << myclock.CpuTime()
-            << " s (cpu time)" << std::endl;
-  std::cout << "Ending the tree merging." << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "Finished endJob after " << myclock.RealTime() << " s (real time) / "
+                                   << myclock.CpuTime() << " s (cpu time)" << std::endl;
+  LogDebug("TkAlCaSkimTreeMerger") << "Ending the tree merging." << std::endl;
   out_->Write();
-  std::cout << "Deleting..." << std::flush;
+  LogDebug("TkAlCaSkimTreeMerger") << "Deleting..." << std::flush;
```
Comment (tvami) [inline Alignment/TrackerAlignment/plugins/TkAlCaSkimTreeMerger.cc:285]: Please remove std::endl and std::flush

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc (hunk 9 above)
Comment (tvami) [reply Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:212]: Or add the `std::`, btw this means that there is an `using namespace std;` in the code, right? I thought that's discouraged

File: Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc (hunk 10 above)
Comment (tvami) [reply Alignment/TrackerAlignment/plugins/AlignmentPrescaler.cc:220]: Or add the `std::`, btw this means that there is an `using namespace std;` in the code, right? I thought that's discouraged

**Review by tvami — COMMENTED** [review]
Cosmetic comments about `std::endl`, `std::flush` and in general about the inclusion of std namespace in the code.

**Comment by mmusich** [comment]
@tvami,
I have tested removal of the `std::endl` at the end of LogDebug statements and found no functional difference in the output. About inclusion of ` using namespace std` I don't find any duscouraging remark here[here](https://cms-sw.github.io/cms_coding_rules.html).
As there is no added value in the comments proposed, I choose to ignore them.

**Comment by tvami** [comment]
> @tvami,
> I have tested removal of the `std::endl` at the end of LogDebug statements and found no functional difference in the output. About inclusion of ` using namespace std` I don't find any discouraging remark [here](https://cms-sw.github.io/cms_coding_rules.html).
> As there is no added value in the comments proposed, I choose to ignore them.

Ok, that's fine, it was all cosmetic.

**Comment by tvami** [comment]
+alca

**Comment by qliphy** [comment]
+1

**Pull request merged**